File: web/src/components/manga/structs.ts
export type Lang = "en" | "jp";

export interface Manga {
  title: LangOptions;
  thumbnail: number[];
  volume_count: number;
}

export interface Volume {
  title: string;
  metadata: VolumeMetadata;
}

export interface VolumeMetadata {
  page_count: number;
}

export interface VolumeList {
  en: Volume[],
  jp: Volume[],
}

export interface Page {
  img: number[],
  metadata: VolumeMetadata,
}

export interface LangOptions {
  en: boolean;
  jp: boolean;
}

export interface Filters {
  langs: LangOptions;
}

export const ImageError = "ERROR";

File: api/src/routes/manga/structs.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct Manga {
    pub title: LangOptions,
    pub volume_count: i32,
    pub thumbnail: Vec<u8>,
}

#[derive(Serialize, Deserialize)]
pub struct Metadata {
    pub title: LangOptions,
    pub volume_count: i32,
    pub thumbnail: String,
}

#[derive(Serialize, Deserialize)]
pub struct Volume {
    pub title: String,
    pub metadata: VolumeMetadata,
}

#[derive(Serialize, Deserialize)]
pub struct VolumeMetadata {
    pub page_count: i32,
}

#[derive(Serialize, Deserialize)]
pub struct VolumeList {
    pub en: Vec<Volume>,
    pub jp: Vec<Volume>,
}

#[derive(Serialize, Deserialize)]
pub struct Page {
    pub img: Vec<u8>,
    pub metadata: VolumeMetadata,
}

#[derive(Serialize, Deserialize)]
pub struct LangOptions {
    pub en: String,
    pub jp: String,
}
